libvast/src/system/index.cpp

```diff
@@ -1020,7 +1004,8 @@ index(index_actor::stateful_pointer<index_state> self,
       if (iter == self->state.pending.end()) {
         VAST_WARN("{} drops query for unknown query id {}", *self, query_id);
         self->send(client, atom::done_v);
-        return {};
+        return caf::make_error(ec::lookup_error, "unknown query id");
```

Comment:
We're returning an error to the client, so why are we sending `done` as well?

In general, I think sending `done` manually and not as part of a reply is the biggest structural issue with this code.

Reply:
We probably don't have to send a done in this case.

The tricky part of this code is that the `query` handler essentially returns 2 messages, the triple and later the `done` for the first batch. The alternative would be to change the protocol so that the client must send an extra message to request the first batch. Tbh I think it's good that we're avoiding that extra roundtrip, and besides, this PR is certainly not the place to change the protocol.

Reply:
Makes sense not to do such a thing in this PR.